### PR TITLE
Issue/16334 link blogging reminders with prompts onboarding

### LINKED ISSUE #16334: [Blogging prompts v1] Link help button of Blogging Reminder Setting dialog with a special version of Onboarding Dialog

### PR DESCRIPTION
**Pull request by khaykov**
Fixes #16334

This PR links adds the little help icon (?) to the blogging prompt switch at the blogging reminder cadence setting screen, that leads to the version of onboarding dialog with only "Got it" button that dismisses the dialog.

I did a bit of refactoring to the dialog to accommodate two different variants of the dialog, and left some comments in the diff.
@renanferrari since you had the vision for original code, please let me know if you want me to do anything in a different way :)

[Image from Gyazo](https://gyazo.com/b95055e85d8953caf681c77bad3363c8)

To test:
- Enable blogging prompt FF.
- Navigate to Settings, Blogging Reminder screen.
- Select a date if not selected already.
- Tap on the ? icon near blogging prompt switch.
- Confirm that the onboarding dialog appears, with only one "Got it" button that dismisses it.
- Confirm that the buttons in regular variant of onboarding dialog work as expected.

## Regression Notes
1. Potential unintended areas of impact
Onboarding version of the dialog shows one button.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual and automated tests.

3. What automated tests I added (or what prevented me from doing so)
Testing click listeners and the Ready state.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/bloggingprompts/onboarding/BloggingPromptsOnboardingUiState.kt [hunk 1]
```diff
@@ -10,6 +10,12 @@ sealed class BloggingPromptsOnboardingUiState {
         @StringRes val contentTopRes: Int,
         @StringRes val contentBottomRes: Int,
         @StringRes val contentNoteTitle: Int,
-        @StringRes val contentNoteContent: Int
+        @StringRes val contentNoteContent: Int,
+        @StringRes val primaryButtonLabel: Int,
+        val isPrimaryButtonVisible: Boolean,
+        val onPrimaryButtonClick: () -> Unit,
+        @StringRes val secondaryButtonLabel: Int,
```
Comment (khaykov) [inline WordPress/src/main/java/org/wordpress/android/ui/bloggingprompts/onboarding/BloggingPromptsOnboardingUiState.kt:17]: Added the listeners to the Ready state alongside other parameters, so we can test VM with real instance of `BloggingPromptsOnboardingUiStateMapper`

File: WordPress/src/test/java/org/wordpress/android/ui/bloggingprompts/onboarding/BloggingPromptsOnboardingUiStateMapperTest.kt [hunk 2]
```diff
@@ -3,72 +3,83 @@ package org.wordpress.android.ui.bloggingprompts.onboarding
...
 import org.wordpress.android.R
+import org.wordpress.android.R.string
 import org.wordpress.android.ui.avatars.TrainOfAvatarsItem.AvatarItem
 import org.wordpress.android.ui.avatars.TrainOfAvatarsItem.TrailingLabelTextItem
+import org.wordpress.android.ui.bloggingprompts.onboarding.BloggingPromptsOnboardingDialogFragment.DialogType.INFORMATION
+import org.wordpress.android.ui.bloggingprompts.onboarding.BloggingPromptsOnboardingDialogFragment.DialogType.ONBOARDING
+import org.wordpress.android.ui.bloggingprompts.onboarding.BloggingPromptsOnboardingUiState.Ready
 import org.wordpress.android.ui.utils.UiString.UiStringPluralRes
 
 class BloggingPromptsOnboardingUiStateMapperTest {
```
Comment (khaykov) [inline WordPress/src/test/java/org/wordpress/android/ui/bloggingprompts/onboarding/BloggingPromptsOnboardingUiStateMapperTest.kt:14]: Testing individual parts of ready state is definitely more precise, but the number of tests grows a lot when new variation of the state is added, so I switched to comparing whole state.

File: WordPress/src/main/java/org/wordpress/android/ui/bloggingprompts/onboarding/BloggingPromptsOnboardingDialogFragment.kt [hunk 3]
```diff
@@ -158,6 +168,9 @@ class BloggingPromptsOnboardingDialogFragment : FeatureIntroductionDialogFragmen
                                 .onSetPromptReminderClick(action.selectedSiteLocalId)
                     }
                 }
+                is DismissDialog -> {
```
Comment (RenanLukas) [inline WordPress/src/main/java/org/wordpress/android/ui/bloggingprompts/onboarding/BloggingPromptsOnboardingDialogFragment.kt:171]: 👍

**Review by RenanLukas — APPROVED** [review]
@khaykov thanks for the PR
LGTM :shipit:

**Pull request merged**